karspexet/ticket/views.py

```diff
@@ -44,36 +44,61 @@ def home(request):
     })
 
 
+@transaction.atomic
 def select_seats(request, show_slug):
     show = Show.objects.get(slug=show_slug)
     reservation = _get_or_create_reservation_object(request, show)
     taken_seats_qs = Reservation.active.exclude(pk=reservation.pk).filter(show=show)
     _set_session_timeout(request)
 
+    seats_in_venue = Seat.objects.filter(group__venue=show.venue).all()
+    available_seats = seats_in_venue.exclude(id__in=show.ticket_set.values_list('seat_id')).all()
+
     if request.POST:
-        seat_params = _seat_specifications(request)
-        if not _all_seats_available(taken_seats_qs, seat_params.keys()):
-            messages.error(request, "Vissa av platserna du valde har redan blivit bokade av någon annan")
-        elif _some_seat_is_missing_ticket_type(seat_params):
-            messages.error(request, "Du måste välja biljettyp för alla platser")
-        else:
-            reservation.tickets = seat_params
-            reservation.save()
-            return redirect("booking_overview", show_slug=show.slug)
+        if show.free_seating:
```

Comment:
We might want to put the different logic between no seating and seating map into two different classes? That way this method wouldn’t be super long and include multiple `if show.free_seating:` :)

Reply:
I completely agree, and I will adress this and some complaints from Codacy before merging this. The current version of this branch is however already deployed to the production server due to some time constraints.

Reply:
On that note, we might want to have a look at converting these views to [Class-based views](https://docs.djangoproject.com/en/2.2/topics/class-based-views/), to easier be able to separate the parts of the view functions into separate methods.